canary/webrtc-c/jobs/runner.groovy

```diff
@@ -272,8 +262,28 @@ pipeline {
         string(name: 'GIT_HASH')
         booleanParam(name: 'FIRST_ITERATION', defaultValue: true)
     }
+    
+    // Set the role ARN to environment to avoid string interpolation to follow Jenkins security guidelines.
+    environment {
+        AWS_KVS_STS_ROLE_ARN = credentials('CANARY_STS_ROLE_ARN')
+    }
 
     stages {
+        stage('Fetch STS credentials and export to env vars') {
+            steps {
+                script {
+                    def assumeRoleOutput = sh(script: 'aws sts assume-role --role-arn $AWS_KVS_STS_ROLE_ARN --role-session-name roleSessionName --duration-seconds 43200 --output json',
+                                                returnStdout: true
+                                                ).trim()
+                    def assumeRoleJson = readJSON text: assumeRoleOutput
+
+                    env.AWS_ACCESS_KEY_ID = assumeRoleJson.Credentials.AccessKeyId
+                    env.AWS_SECRET_ACCESS_KEY = assumeRoleJson.Credentials.SecretAccessKey
+                    env.AWS_SESSION_TOKEN = assumeRoleJson.Credentials.SessionToken
+                }
+            }
+        }
+
```

Comment:
if we have post step, let us clear the env

env.AWS_ACCESS_KEY_ID = ''
            env.AWS_SECRET_ACCESS_KEY = ''
            env.AWS_SESSION_TOKEN = ''